include/flexflow/model.h

```diff
@@ -267,6 +267,25 @@ class Replicate;
 class FusedParallelOp;
 class ParallelOpInfo;
 
+// TODO: Move to an appropriate place
```

Comment:
This looks good, but a few brief comments explaining how it works (i.e. what each piece does) for people not as familiar with template metaprogramming would be a good idea